QMWP.php

```diff
@@ -1107,6 +1109,58 @@ private function get_plugin_template_html($shortCodeName, $version)
         return $templateContent;
     }
 
+    /**
+     * Will add script with JS alert if variable is null or undefined
+     *
+     * call: add_null_variable_alert(array(false => 'Ops!'), '<div></div>');
+     * will return: <div><script>alert("Ops!");</script></div>
+     */
+    private function add_null_global_variable_alerts($variables, $templateContent)
+    {
+        $alertsHtmlString = "<script id='null-global-vars-alerts'>\n";
+        foreach ($variables as $value => $message) {
+
+            if (empty($value)) {
+                $alertsHtmlString .= "alert('" . $message . "');\n";
+            }
+
+        }
+        $alertsHtmlString .= "</script>\n";
+
+        return $alertsHtmlString .= $templateContent;
+
+    }
+
+    /**
+     * Do some decoration of a template before returning HTML
+     * Currently it sets global JS variables anb adds an alerts
+     *
+     * @param $template
+     * @return string
+     */
+    private function process_template($template)
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [process_template accesses the super-global variable $_SERVER.](https://www.codacy.com/app/m_3/QuantiModo-WordPress-Plugin/file/964742200/issues/source?bid=2836188&fileBranchId=2859489#l1141)